include/flatbuffers/flatbuffers.h

```diff
@@ -123,6 +125,31 @@
   #define FLATBUFFERS_NOEXCEPT
 #endif
 
+// NOTE: the FLATBUFFERS_DELETE_FUNC macro may change the access mode to
+// private, so be sure to put it at the end or reset access mode explicitly.
+#if (!defined(_MSC_VER) || _MSC_FULL_VER >= 180020827) && \
+    (!defined(__GNUC__) || (__GNUC__ * 100 + __GNUC_MINOR__ >= 404))
+  #define FLATBUFFERS_DELETE_FUNC(func) func = delete;
+#else
+  #define FLATBUFFERS_DELETE_FUNC(func) private: func;
+#endif
+
+#if (__cplusplus >= 201402L) && defined(__has_cpp_attribute)
+  #if __has_cpp_attribute(deprecated)
+    #define DEPRECATED(msg, func) [[deprecated(msg)]] func
```

Comment:
Okay, I just thought it'd make upgrades easier if there was a compiler warning/error telling you what changed. But we can certainly axe things aggressively instead.